pageserver/src/tenant/timeline/init.rs

```diff
@@ -99,14 +98,12 @@ impl LocalLayerFileMetadata {
 pub(super) enum Decision {
     /// The layer is not present locally.
     Evicted(LayerFileMetadata),
-    /// The layer is present locally, but local metadata does not match remote; we must
-    /// delete it and treat it as evicted.
-    UseRemote {
+    /// The layer is present locally, and metadata matches: we may hook up this layer to the
+    /// existing file in local storage.
+    UseLocal {
```

Comment:
Not sure if I understand what does this comment mean, or why is UseRemote removed.

Reply:
Previously, we had UseLocal, which meant "use remote metadata, and the local layer exists", and UseRemote, whichc meant "use remote metadata, and the local layer exists but doesn't match and should be deleted".

In this PR I've changed Decision to just reflect whether the layer is resident or not, and for the case where we want to delete the local file, I've moved that to DismissedLayer::BadMetadata.  This simplifies the logic, because a Decision now always refers to a layer that exists in remote storage.

Reply:
The naming was a bit weird though: I've changed this to call the variants Evicted and Resident instead of Evicted and UseLocal.